layer/nrlf/nrlf/core/repository.py

```diff
@@ -525,7 +525,15 @@ def _delete(id):
         if len(delete_pks) >= MAX_TRANSACT_ITEMS:
             raise TooManyItemsError("Too many items to process in one transaction")
         transact_items = [_delete(id) for id in delete_pks] + [_put()]
-        return self.dynamodb.transact_write_items(TransactItems=transact_items)
+        try:
+            return self.dynamodb.transact_write_items(TransactItems=transact_items)
+        except ClientError as error:
+            if "CancellationReasons" in error.response:
+                reasons = error.response["CancellationReasons"]
+                # Last reason relates to the single PUT and indicates duplicate item
+                if reasons[-1]["Code"] == "ConditionalCheckFailed":
+                    raise DuplicateError("Condition check failed - Duplicate item")
+            raise error
```

Comment:
to avoid the comment could wrap this up in a context manager function

```python
with handler_put_duplicate_error(put_idx=put_idx):
    return self.dynamodb.transact_write_items(TransactItems=transact_items)
```

Reply:
We can also search through the `Item` on the `reasons` to see if it's `id` matches the subject's `id`.  This will be simpler than the index additions above.